docs/porting.md

```diff
@@ -369,11 +412,10 @@ class SafeCounter:
         self.lock = threading.Lock()
 
     def increment(self):
-        self.lock.acquire()
-        current_value = self.value
-        time.sleep(random.randint(0, 10) * 0.0001)
-        self.value = current_value + 1
-        self.lock.release()
+        with self.lock:
+            current_value = self.value
+            time.sleep(random.randint(0, 10) * 0.0001)
+            self.value = current_value + 1
```

Comment:
_technically_ the code between `acquire()` and `release()` can't raise.
However, I would much rather treat the previous version as an antipattern as a matter of principle, as it relieves you from having to think if you're introducing a possibility of exception every time you maintain the code.

Reply:
I agree it's better to do it with the context manager.